### PR TITLE
add some logging info

### PR DESCRIPTION
**Pull request by nhayfield**
## Summary
Shows logs on the view page

## Checklist

- [ ] reference any related issues
- [ ] updated docs
- [ ] updated unit tests
- [ ] updated UPGRADING.md
- [ ] add appropriate tag (`improvement` / `bug` / etc)
- [x] ready for review

### CODE REVIEW

File: src/main.dev.ts [hunk 1]
```diff
@@ -289,22 +291,27 @@ app.on('ready', async () => {
...
-      stream.on('data', (response) => {
+      updateStream.on('data', (response) => {
         sendTo?.send(LISTENER_LOG, {
-          msg: response.message,
+          msg: response as ConnectionStatusUpdate,
+          remoteAddr: args.remoteAddr,
         });
       });
+      // empty function otherwise causes fatal error on cancel !!!
+      updateStream.on('error', () => {});
```
Comment (wasaga) [inline src/main.dev.ts:307]: `console.error` maybe?

File: src/renderer/components/Toast.tsx [hunk 2]
```diff
@@ -2,20 +2,29 @@ import * as React from 'react';
...
 
   useEffect(() => {
+    mounted.current = true;
+
     setTimeout(() => {
-      setOpen(false);
+      if (mounted.current) {
+        setOpen(false);
+      }
     }, 6000);
```
Comment (wasaga) [inline src/renderer/components/Toast.tsx:23]: maybe we should have a central place for all such UI constants?

File: src/renderer/pages/ConnectionView.tsx [hunk 3]
```diff
@@ -77,21 +105,77 @@ const ConnectionView = (): JSX.Element => {
...
+      case 1:
+        message = msg.peerAddr + ' opening connection to ' + remoteAddr;
+        break;
+      case 2:
+        message =
+          msg.peerAddr +
+          ' authentication with ' +
+          msg.authUrl +
+          ' required for ' +
+          connection.remoteAddr;
```
Comment (wasaga) [inline src/renderer/pages/ConnectionView.tsx:136]:
```suggestion
          remoteAddr;
```

as there's no `remoteAddr` property for `connection` and it causes `undefined`

image

File: src/renderer/pages/ConnectionView.tsx [hunk 4]
```diff
@@ -77,21 +105,77 @@ const ConnectionView = (): JSX.Element => {
...
+        break;
+      case 3:
+        message = msg.peerAddr + ' connected to ' + remoteAddr;
+        break;
+      case 4:
+        message = msg.peerAddr + ' disconnected from ' + remoteAddr;
```
Comment (wasaga) [inline src/renderer/pages/ConnectionView.tsx:142]: I'm not sure if repetitive `remoteAddr` mention is justified - logs are only visible for a particular connection where you already see the remote address. 

image

@travisgroth

**Review by wasaga — CHANGES_REQUESTED** [review]
pls fix `undefined` - see suggestion

File: src/main.dev.ts (hunk 1 above)
Comment (nhayfield) [reply src/main.dev.ts:307]: no this is a hack to fix a library bug. we dont want the console to show anything

File: src/renderer/components/Toast.tsx (hunk 2 above)
Comment (nhayfield) [reply src/renderer/components/Toast.tsx:23]: meh

File: src/renderer/pages/ConnectionView.tsx (hunk 3 above)
Comment (nhayfield) [reply src/renderer/pages/ConnectionView.tsx:136]: yeah missed this obe when i refactored

File: src/renderer/pages/ConnectionView.tsx (hunk 4 above)
Comment (nhayfield) [reply src/renderer/pages/ConnectionView.tsx:142]: agreed we could remove them

**Review by wasaga — APPROVED** [review]

**Pull request merged**